tests/test_sqlite_adapter.py

```diff
@@ -0,0 +1,249 @@
+import platform
+import sqlite3
+import sys
+from datetime import datetime, timezone
+
+import pytest
+
+from flow.record import Record, RecordDescriptor, RecordReader, RecordWriter
+from flow.record.adapter.sqlite import sanitized_name
+
+
+def generate_records(amount) -> Record:
+    """Generates some test records"""
+    TestRecordWithFooBar = RecordDescriptor(
+        "test/record",
+        [
+            ("string", "name"),
+            ("string", "foo"),
+            ("string", "bar"),
+        ],
+    )
+    for i in range(amount):
+        yield TestRecordWithFooBar(name=f"record{i}", foo="bar", bar="baz")
+
+
+@pytest.mark.parametrize(
+    "table_name",
+    [
+        "my-movies",
+        "movies",
+        "123213",
+        "_my_movies",
+    ],
+)
+def test_table_name_sanitization(tmp_path, table_name):
+    """Ensure that we can read table names that are technically invalid in flow.record."""
+    db = tmp_path / "records.db"
+    con = sqlite3.connect(db)
+    con.execute(f"CREATE TABLE '{table_name}' (title TEXT, year INTEGER, score REAL)")
+    data = [
+        ("Monty Python Live at the Hollywood Bowl", 1982, 7.9),
+        ("Monty Python's The Meaning of Life", 1983, 7.5),
+        ("Monty Python's Life of Brian", 1979, 8.0),
+    ]
+    con.executemany(f"INSERT INTO '{table_name}' VALUES(?, ?, ?)", data)
+    con.commit()
+    con.close()
+
+    data_records = []
+    with RecordReader(f"sqlite://{db}") as reader:
+        data_records = [(record.title, record.year, record.score) for record in reader]
+    assert data == data_records
+
+
+@pytest.mark.parametrize(
+    "field_name",
+    [
+        "normal_field_name",
+        "_starting_with_underscore",
+        "my-field-name",
+        "1337_starting_with_number",
+    ],
+)
+def test_field_name_sanitization(tmp_path, field_name):
+    """Ensure that we can read field names that are technically invalid in flow.record."""
+    db = tmp_path / "records.db"
+    con = sqlite3.connect(db)
+    con.execute(f"CREATE TABLE 'my_table' ('{field_name}' TEXT)")
+    data = [
+        ("hello",),
+        ("world",),
+        ("good",),
+        ("bye",),
+    ]
+    con.executemany("INSERT INTO 'my_table' VALUES(?)", data)
+    con.commit()
+    con.close()
+
+    data_records = []
+    sanitized_field_name = sanitized_name(field_name)
+
+    with RecordReader(f"sqlite://{db}") as reader:
+        data_records = [(getattr(record, sanitized_field_name),) for record in reader]
+    assert data == data_records
+
+
+@pytest.mark.skipif(
+    (platform.python_implementation() == "PyPy" and sys.version_info[:2] == (3, 9)),
+    reason="PyPy 3.9 seems to have issues with SQLite transactions",
```

Comment:
Interesting :smile: do you not want to guard or warn about this in the actual writer?

Reply:
This actually made me investigate the problem a bit more, and it seems it was caused by WAL mode. I've just disabled it for now.

Reply:
Tests are now passing for PyPy 3.9 after removing `self.con.execute("PRAGMA journal_mode='wal';")`